### PR TITLE
refactor(tests): test helper functions for messaging and meet tests

### PR DESCRIPTION
**Pull request by moriahmaney**

### CODE REVIEW

File: test/journeys/.eslintrc.yml [hunk 1]
```diff
@@ -5,6 +5,8 @@ rules:
     - 0
   global-require:
     - 0
+  max-nested-callbacks:
+    - 0
```
Comment (adamweeks) [inline test/journeys/.eslintrc.yml]: 👍

File: test/journeys/lib/test-helpers/space-widget/meet.js [hunk 2]
```diff
@@ -57,3 +63,101 @@ export function hangup(aBrowser) {
...
   aBrowser.element(elements.meetWidget).element(elements.hangupButton).click();
 }
+
+/**
+ * Test to hangup call before reciever answers
+ * @param {Object} browserLocal
+ * @param {Object} browserRemote
+ * @returns {void}
+ */
+export function hangupBeforeTest(browserLocal, browserRemote) {
```
Comment (adamweeks) [inline test/journeys/lib/test-helpers/space-widget/meet.js]: `hangupBeforeAnswerTest` seems more appropriate

File: test/journeys/lib/test-helpers/space-widget/meet.js [hunk 3]
```diff
@@ -57,3 +63,101 @@ export function hangup(aBrowser) {
...
+ */
+export function hangupDuringTest(browserLocal, browserRemote) {
+  clearEventLog(browserLocal);
+  clearEventLog(browserRemote);
+  switchToMeet(browserLocal);
+  call(browserLocal, browserRemote);
+  answer(browserRemote);
+  hangup(browserLocal);
+  browserLocal.waitForVisible(elements.messageWidget);
+  const result = browserLocal.execute(() => {
```
Comment (adamweeks) [inline test/journeys/lib/test-helpers/space-widget/meet.js]: Could you move these call event parsing code blocks into their own function. Looks like it is copy pasted in the `callEventTest` function.

Comment (adamweeks) [inline test/journeys/lib/test-helpers/space-widget/meet.js]: You could modify the `getEventLog` in `events.js` to only delete that call data object if it exists, that way we could share that function across all event tests.

File: test/journeys/specs/tap/widget-message-meet/index.js [hunk 4]
```diff
@@ -8,12 +8,12 @@ import '@ciscospark/plugin-phone';
 import {switchToMeet, switchToMessage} from '../../../lib/test-helpers/space-widget/main';
 import {clearEventLog, getEventLog} from '../../../lib/events';
 import {sendMessage, verifyMessageReceipt} from '../../../lib/test-helpers/space-widget/messaging';
-import {elements} from '../../../lib/test-helpers/space-widget/meet';
+// import {elements} from '../../../lib/test-helpers/space-widget/meet';
```
Comment (adamweeks) [inline test/journeys/specs/tap/widget-message-meet/index.js]: ?

File: test/journeys/lib/test-helpers/space-widget/meet.js (hunk 2 above)
Comment (moriahmaney) [inline test/journeys/lib/test-helpers/space-widget/meet.js]: Yeah, you're right. Want me to change `hangupDuringTest` and `declineIncomingTest` to `hangupDuringCallTest` and `declineIncomingTest` or are those okay?

File: test/journeys/lib/test-helpers/space-widget/messaging.js [hunk 5]
```diff
@@ -16,50 +17,280 @@ export const elements = {
...
+};
+
+/**
+ * Test for sending markdown message with a horizontal line
+ * @param {TestObject} sender
+ * @param {TestObject} receiver
+ * @returns {void}
+ */
+const hr = (sender, receiver) => {
+  // Remote is now ready, send a message from it
```
Comment (moriahmaney) [inline test/journeys/lib/test-helpers/space-widget/messaging.js]: Good catch! Meant to remove those and then forgot 😊

File: test/journeys/specs/tap/widget-message-meet/index.js (hunk 4 above)
Comment (moriahmaney) [inline test/journeys/specs/tap/widget-message-meet/index.js]: ? indeed. removed this line :)

File: test/journeys/lib/test-helpers/space-widget/meet.js (hunk 2 above)
Comment (adamweeks) [inline test/journeys/lib/test-helpers/space-widget/meet.js]: Yea, I think those changes would be beneficial too.

**Comment by adamweeks** [comment]
I updated this to point to the `develop` branch, FYI.

**Pull request merged**